validator/sawtooth_validator/execution/context_manager.py

```diff
@@ -210,112 +272,120 @@ def commit_context(self, context_id_list, virtual):
         return [(a, f.result())
                 for a, f in context.get_from_prefetched(address_list)]
 
     def set(self, context_id, address_value_list):
-        """
-        speculatively sets addresses to a value,
-        can be destroyed or committed to the merkle store
+        """Within a context, sets addresses to a value.
+
         Args:
             context_id (str): the context id returned by create_context
             address_value_list (list): list of {address: value} dicts
 
-        Returns (boolean): True, or False whether the
-
+        Returns:
+            (bool): True if the operation is successful, False if
+                the context_id doesn't reference a known context.
         """
-        with self._shared_lock:
-            if context_id not in self._contexts:
-                LOGGER.info("Context_id not in contexts, %s", context_id)
-                return False
-        with self._shared_lock:
-            context = self._contexts.get(context_id)
-            context.can_set(address_value_list)
-            add_value_dict = {}
-            for d in address_value_list:
-                for add, val in d.items():
-                    add_value_dict[add] = val
-            context.set_futures(add_value_dict)
+
+        if context_id not in self._contexts:
+            LOGGER.warning("Context_id not in contexts, %s", context_id)
+            return False
+
+        context = self._contexts.get(context_id)
+        context.can_set(address_value_list)
+        add_value_dict = {}
+        for d in address_value_list:
+            for add, val in d.items():
+                add_value_dict[add] = val
+        context.set_futures(add_value_dict)
         return True
 
     def get_squash_handler(self):
         def _squash(state_root, context_ids):
             tree = MerkleDatabase(self._database, state_root)
             updates = dict()
             for c_id in context_ids:
-                with self._shared_lock:
-                    context = self._contexts[c_id]
-                for add in context.get_address_value_dict().keys():
+                context = self._contexts[c_id]
+                for add in context.get_state().keys():
                     if add in updates:
                         raise SquashException(
                             "Duplicate address {} in context {}".format(
                                 add, c_id))
 
                 effective_updates = {}
-                for k, val_fut in context.get_address_value_dict().items():
+                for k, val_fut in context.get_state().items():
                     value = val_fut.result()
                     if value is not None:
                         effective_updates[k] = value
 
                 updates.update(effective_updates)
 
             state_hash = tree.update(updates, virtual=False)
+            # clean up all contexts that are involved in being squashed.
+            base_c_ids = []
+            for i in context_ids:
```

Comment:
Changing L384 right now. I also found another place where I used an i for a variable name in a loop. There is no i in team, and there won't be i's in the diff either.